source/ocsp-support/ocsp-support.rst

```diff
@@ -134,28 +136,69 @@ Drivers SHOULD validate OCSP Responses in the manner specified in `RFC
 6960: 3.2 <https://tools.ietf.org/html/rfc6960#section-3.2>`__ to the
 extent that their TLS library allows.
 
+MongoClient Configuration
+--------------------------
+
+This specification introduces the client-level configuration option
+defined below. Drivers that can, on a per MongoClient basis, disable
+non-stapled OCSP while keeping stapled OCSP enabled MUST implement
+this option.
+
+tlsDisableOcspEndpointCheck
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This boolean option determines whether MongoClient should refrain from
+reaching out to an OCSP endpoint i.e.  whether non-stapled OCSP should
+be disabled.  When set to true, a driver MUST NOT reach out to OCSP
+endpoints. When set to false, a driver MUST reach out to xOCSP
+endpoints if needed (as described in :ref:`Suggested OCSP Behavior`).
+This option MUST default to false.
+
+Naming Deviations
+^^^^^^^^^^^^^^^^^^
+
+Drivers MUST use the defined name of ``tlsDisableOcspEndpointCheck``
+for the connection string parameter to ensure portability of
+connection strings across applications and drivers. If drivers solicit
+MongoClient options through another mechanism (e.g. an options
+dictionary provided to the MongoClient constructor), drivers SHOULD
+use the defined name but MAY deviate to comply with their existing
+conventions. For example, a driver may use
+``tls_disable_ocsp_endpoint_check`` instead of
+``tlsDisableOcspEndpointCheck``.
+
 How OCSP interacts with existing configuration options
 ------------------------------------------------------
 
 The following requirements apply only to drivers that are able to
 enable/disable OCSP on a per MongoClient basis.
 
-1. If a connection string specifies `tlsInsecure=true` then the driver
-   MUST disable OCSP.
+1. If a connection string specifies ``tlsInsecure=true`` then the
+   driver MUST disable OCSP.
+
+2. If a connection string contains both ``tlsInsecure`` and
```

Comment:
We have a similar requirement for `tlsInsecure` and `tlsAllowInvalidCertificates`.

Reply:
These don't appear in conflict to me, since disabling OCSP endpoint checks also reduces security.

Reply:
Hm, I don't think the options are conflict, but I was also going off the precedent set by tlsInsecure and tlsAllowInvalidCertificates:  wouldn't setting both tlsInsecure=true and tlsAllowInvalidCertificates=true reduce security? 

My reading of https://github.com/mongodb/specifications/blob/master/source/uri-options/uri-options.rst#conflicting-tls-options is that setting both these options isn't allowed despite the fact that tlsInsecure is a superset of tlsAllowInvalidCertificates (and thus setting them both to true shouldn't conflict). @saghm Any thoughts?

Reply:
I think it would be consistent with the existing rules for `tlsAllowInvalidCertificates` and `tlsAllowInvalidHostnames` to forbid `tlsDisableOcspEndpointCheck` and `tlsInsecure` from both being present. The consensus we came to when we were writing the URI options spec was that there isn't any utility in specifying `tlsInsecure` along with something like `tlsAllowInvalidCertificates` due to the fact that the latter will be redundant if the two are consistent and confusing regardless of the semantics we pick if they aren't consistent; I think that applies here as well.

Reply:
I relent.